docs/manuals/gui/viselements/charts/heatmap.md

```diff
@@ -0,0 +1,223 @@
+- For marketing goals and analytics, reflecting on user behavior on specific webpages.
+- And more
+
+### Basic Heatmap
+The chart below shows the colors in accordance with each temperatures
+
+```py
+from taipy.gui import Gui
+
+data = {
+    "temperature": [[1, 20, 30], [20, 1, 60], [30, 60, 1]],
+}
+options = {'colorscale': 'Reds'}
+md = """
+## Basic Heatmap
+<|{data}|chart|z=temperature|type=heatmap|options={options}|>
+"""
+
+Gui(md).run()
+```
+![Basic Heatmap](heatmap_basic.png)
+
+### Heatmap with Categorical Axis Labels
+In the following example, we have an array which defines the data (harvest by different farmers in tons/year) to color code. We then also need two lists of names of farmers and vegetables cultivated by them.
+```py
+from taipy.gui import Gui
+import pandas as pd
+
+data = [
+    pd.DataFrame({
+        'harvest': [
+            [0.8, 2.4, 2.5, 3.9, 0.0, 4.0, 0.0],
+            [2.4, 0.0, 4.0, 1.0, 2.7, 0.0, 0.0],
+            [1.1, 2.4, 0.8, 4.3, 1.9, 4.4, 0.0],
+            [0.6, 0.0, 0.3, 0.0, 3.1, 0.0, 0.0],
+            [0.7, 1.7, 0.6, 2.6, 2.2, 6.2, 0.0],
+            [1.3, 1.2, 0.0, 0.0, 0.0, 3.2, 5.1],
+            [0.1, 2.0, 0.0, 1.4, 0.0, 1.9, 6.3]
+        ],
+        'vegetables': ["cucumber",
+                       "tomato",
+                       "lettuce",
+                       "asparagus",
+                       "potato",
+                       "wheat",
+                       "barley"]
+    }),
+    pd.DataFrame({
+        'farmers': ["Farmer Joe",
+                    "Upland Bros.",
+                    "Smith Gardening",
+                    "Agrifun",
+                    "Organiculture",
+                    "BioGoods Ltd.",
+                    "Cornylee Corp."],
+    })
+]
+options = {'colorscale': 'Rainbow'}
+md = """
+## Heatmap with Categorical Axis Labels
+<|{data}|chart|type=heatmap|z=0/harvest|x=1/farmers|y=0/vegetables|options={options}|>
+"""
+
+Gui(md).run()
+```
+![Heatmap with Categorical Axis Labels](heatmap_axis_labels.png)
+
+### Annotated Heatmap
+In this chart, we have average temperature (°F) in seasons of top 4 US cities
+```py
+from taipy.gui import Gui
+import pandas as pd
+
+seasons = ['Winter', 'Summer', 'Spring', 'Autumn']
+
+cities = ['New York', 'Los Angeles', 'Chicago', 'Houston']
+
+temperature = [
+    [33.7, 74.49, 57.8, 57.6],
+    [60.1, 68.2, 64.5, 65.7],
+    [22.89, 72.2, 55.7, 51.6],
+    [53.0, 83.3, 72.7, 53.0],
+]
+
+data = [
+    pd.DataFrame({
+        'temperature': temperature,
+        'cities': cities
+    }),
+    pd.DataFrame({
+        'seasons': seasons,
+    })
+]
+
+options = {'colorscale': 'Portland'}
+
+layout = {
+    'annotations': [],
+    'xaxis': {
+        'ticks': '',
+        'side': 'top'
+    },
+    'yaxis': {
+        'ticks': '',
+        'ticksuffix': ' ',
+        'width': 700,
+        'height': 700,
+        'autosize': False
+    }
+}
+
+for i in range(len(cities)):
```

Comment:
I suspect that this could use the Python list comprehension, but maybe it would be far less readable.

Reply:
It is great If you can give me an example. I am not familiar with Python, the way I did is the result from Google :)

Reply:
No no... that's fine. Because you need the city/season index you have little choice but to code this the way you did.
Thx.